evidence/cli/command_cli.go

```diff
@@ -70,15 +70,42 @@ func validateCreateEvidenceCommonContext(ctx *components.Context) error {
 	if !ctx.IsFlagSet(predicate) || assertValueProvided(ctx, predicate) != nil {
 		return errorutils.CheckErrorf("'predicate' is a mandatory field for creating evidence: --%s", predicate)
 	}
+
 	if !ctx.IsFlagSet(predicateType) || assertValueProvided(ctx, predicateType) != nil {
 		return errorutils.CheckErrorf("'predicate-type' is a mandatory field for creating evidence: --%s", predicateType)
 	}
-	if !ctx.IsFlagSet(key) || assertValueProvided(ctx, key) != nil {
-		return errorutils.CheckErrorf("'key' is a mandatory field for creating evidence: --%s", key)
+
+	if err := ensureKeyExists(ctx, key); err != nil {
+		return err
+	}
+
+	if !ctx.IsFlagSet(keyAlias) {
+		setKeyAliasIfNeeded(ctx, keyAlias)
 	}
+
 	return nil
 }
 
+func ensureKeyExists(ctx *components.Context, key string) error {
+	if ctx.IsFlagSet(key) && assertValueProvided(ctx, key) == nil {
+		return nil
+	}
+
+	signingKeyValue, _ := getEnvVariable(evdSigningKey)
+	if signingKeyValue == "" {
+		return errorutils.CheckErrorf("'key' or EVD_KEY must be provided when creating evidence: --%s", key)
+	}
+	ctx.AddStringFlag(key, signingKeyValue)
+	return nil
+}
+
+func setKeyAliasIfNeeded(ctx *components.Context, keyAlias string) {
+	evdKeyAliasValue, _ := getEnvVariable(evdKeyAlias)
```

Comment:
Maybe the keyalias was provided as a parameter and the key was read from the environment variable

Reply:
if the key was provided as a param - we won't get here; the code in line 82 checks
if !ctx.IsFlagSet(keyAlias) {
       setKeyAliasIfNeeded(ctx, keyAlias)
}
if the key was provided as param - we do nothing and the param value will be taken. if not - we will check if we have env var - if so - we will use it.

Reply:
Then change the function name, if you call `setKeyAliasIfNeeded` you indeed need alias.
By the way it still not guaranteed to have such thing.